aw_server/api.py

```diff
@@ -1,442 +1,180 @@
+                  self.db[bucket_id].get(limit, start, end)]
         return events
 
-    @api.expect(event)
-    def post(self, bucket_id):
-        """
-        Create events for a bucket. Can handle both single events and multiple ones.
-        """
-        data = request.get_json()
-        logger.debug("Received post request for event in bucket '{}' and data: {}".format(bucket_id, data))
-
-        if bucket_id not in app.db.buckets():
-            msg = "There's no bucket named {}".format(bucket_id)
-            raise BadRequest("NoSuchBucket", msg)
-
-        if isinstance(data, dict):
-            events = [Event(**data)]
-        elif isinstance(data, list):
-            events = [Event(**e) for e in data]
-        else:
-            raise BadRequest("Invalid POST data", "")
+    @check_bucket_exists
+    def create_events(self, bucket_id: str, events: List[Event]):
+        """Create events for a bucket. Can handle both single events and multiple ones."""
+        self.db[bucket_id].insert(events)
+        return None
 
-        app.db[bucket_id].insert(events)
-        return {}, 200
-
-
-@api.route("/0/buckets/<string:bucket_id>/events/replace_last")
-class ReplaceLastEventResource(Resource):
-    """
-    Replaces last event inserted into bucket
-    """
-
-    @api.expect(event)
-    def post(self, bucket_id):
+    @check_bucket_exists
+    def heartbeat(self, bucket_id: str, heartbeat: Event, pulsetime: float) -> Event:
         """
-        Replace last event inserted into the bucket
-        """
-        logger.debug("Received post request for event in bucket '{}' and data: {}".format(bucket_id, request.get_json()))
-
-        if bucket_id not in app.db.buckets():
-            msg = "There's no bucket named {}".format(bucket_id)
-            raise BadRequest("NoSuchBucket", msg)
-        data = request.get_json()
-
-        if not isinstance(data, dict):
-            logger.error("Invalid JSON object")
-            raise BadRequest("InvalidJSON", "Invalid JSON object")
-
-        app.db[bucket_id].replace_last(Event(**data))
-        return {}, 200
-
-
-@api.route("/0/buckets/<string:bucket_id>/heartbeat")
-class HeartbeatResource(Resource):
-    """
-    Heartbeats are useful when implementing watchers that simply keep
-    track of a state, when it's a certain value and when it changes.
-
-    Heartbeats are essentially events without durations.
-
-    If the heartbeat was identical to the last (apart from timestamp), then the last event has its duration updated.
-    If the heartbeat differed, then a new event is created.
+        Heartbeats are useful when implementing watchers that simply keep
+        track of a state, how long it's in that state and when it changes.
 
-    Such as:
-     - Active application and window title (aw-watcher-window)
-     - Active browser tab (aw-watcher-web)
-     - Currently open document (wakatime)
+        Heartbeats are essentially events without durations.
```

Comment:
Confusing explanation since multiple heartbeats does represent a event with a duration.
I would just remove this sentance since the following sentances describes it properly.